daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -2167,6 +2215,30 @@ func (devices *DeviceSet) cancelDeferredRemoval(info *devInfo) error {
 	return err
 }
 
+func (devices *DeviceSet) _cancelDeferredRemoval(info *devInfo) error {
+	logrus.Debugf("devmapper: _cancelDeferredRemoval START(%s)", info.Name())
+	defer logrus.Debugf("devmapper: _cancelDeferredRemoval END(%s)", info.Name())
+
+	var err error
+
+	// Cancel deferred remove
+	for i := 0; i < 100; i++ {
+		err = devicemapper.CancelDeferredRemove(info.Name())
+		if err != nil {
+			if err == devicemapper.ErrBusy {
+				// If we see EBUSY it may be a transient error,
+				// sleep a bit a retry a few times.
+				devices.Unlock()
```

Comment:
If were unlocking the devices lock won't other threads be able to do other unexpected things with this `*devInfo`?